File: src/server/Mimirorg.TypeLibrary/Models/Client/TransportLibCm.cs
using Mimirorg.TypeLibrary.Enums;

namespace Mimirorg.TypeLibrary.Models.Client
{
    public class TransportLibCm
    {
        public string Id { get; set; }
        public string ParentName { get; set; }
        public string ParentIri { get; set; }
        public string Name { get; set; }
        public string Version { get; set; }
        public string FirstVersionId { get; set; }
        public string Iri { get; set; }
        public ICollection<string> ContentReferences { get; set; }
        public string RdsCode { get; set; }
        public string RdsName { get; set; }
        public string PurposeName { get; set; }
        public Aspect Aspect { get; set; }
        public State State { get; set; }
        public int CompanyId { get; set; }
        public string TerminalId { get; set; }
        public TerminalLibCm Terminal { get; set; }
        public string Description { get; set; }
        public DateTime Created { get; set; }
        public string CreatedBy { get; set; }
        public ICollection<AttributeLibCm> Attributes { get; set; }
        public string Kind => nameof(TransportLibCm);
    }
}

File: src/client/src/models/tyle/client/transportLibCm.ts
import { Aspect } from "../enums/aspect";
import { AttributeLibCm } from "./attributeLibCm";
import { TerminalLibCm } from "./terminalLibCm";

export interface TransportLibCm {
  id: string;
  parentName: string;
  parentIri: string;
  name: string;
  version: string;
  firstVersionId: string;
  iri: string;
  contentReferences: string[];
  rdsCode: string;
  rdsName: string;
  purposeName: string;
  aspect: Aspect;
  companyId: number;
  terminalId: string;
  terminal: TerminalLibCm;
  description: string;
  created: string;
  createdBy: string;
  attributes: AttributeLibCm[];
  kind: string;
}
